packages/webapp/src/routes/credits-purchase/credits-purchase-order.js

```diff
@@ -0,0 +1,73 @@
+import constants from '../../utils/constants.js'
+import { checked } from '../../utils/helpers.js'
+
+export default [
+  {
+    method: 'GET',
+    path: constants.routes.CREDITS_PURCHASE_ORDER,
+    options: {
+      auth: false
+    },
+    handler: (request, h) => {
+      const willPOInUse = request.yar.get(constants.redisKeys.WILL_CREDITS_PO_IN_USE)
+      const purchaseOrderNumber = request.yar.get(constants.redisKeys.CREDITS_PURCHASE_ORDER_NUMBER)
+
+      return h.view(constants.views.CREDITS_PURCHASE_ORDER, {
+        willPOInUse,
+        purchaseOrderNumber,
+        checked
+      })
+    }
+  },
+  {
+    method: 'POST',
+    path: constants.routes.CREDITS_PURCHASE_ORDER,
+    options: {
+      auth: false
+    },
+    handler: async (request, h) => {
+      const willPOInUse = request.payload.willPOInUse
+      const purchaseOrderNumber = request.payload.purchaseOrderNumber
+      request.yar.set(constants.redisKeys.WILL_CREDITS_PO_IN_USE, willPOInUse)
+
+      const error = validateData(willPOInUse, purchaseOrderNumber)
+      if (error) {
+        return h.view(constants.views.CREDITS_PURCHASE_ORDER, {
+          purchaseOrderNumber,
+          willPOInUse,
+          checked,
+          ...error
+        })
+      } else if (willPOInUse === 'yes') {
+        request.yar.set(constants.redisKeys.CREDITS_PURCHASE_ORDER_NUMBER, purchaseOrderNumber)
+      } else {
+        request.yar.clear(constants.redisKeys.CREDITS_PURCHASE_ORDER_NUMBER)
+      }
+
+      return h.redirect(constants.routes.CREDITS_APPLICATION_BY_INDIVIDUAL_OR_ORGANISATION, {
+        purchaseOrderNumber: null,
+        willPOInUse,
+        checked
+      })
+    }
+  }
+]
+
+const validateData = (willPOInUse, purchaseOrderNumber) => {
```

Comment:
We should an an extra validation to make sure the purchase order is not longer than 50 characters. 

there is a `validateLengthOfCharsLessThan50` function in `utils/helpers.js` that is used on the registration journey, so we can reuse that